src/util/log/fd_log.h

```diff
@@ -155,6 +157,55 @@
 #define FD_LOG_ALERT(a)   do { long _fd_log_msg_now = fd_log_wallclock(); fd_log_private_2( 6, _fd_log_msg_now, __FILE__, __LINE__, __func__, fd_log_private_0 a ); } while(0)
 #define FD_LOG_EMERG(a)   do { long _fd_log_msg_now = fd_log_wallclock(); fd_log_private_2( 7, _fd_log_msg_now, __FILE__, __LINE__, __func__, fd_log_private_0 a ); } while(0)
 
+#define FD_LOG_HEXDUMP_BYTES_PER_LINE             16
+#define FD_LOG_HEXDUMP_BLOB_DESCRIPTION_MAX_LEN   32
+#define FD_LOG_HEXDUMP_MAX_INPUT_BLOB_SZ          1500
+
+#define FD_LOG_HEXDUMP_ADD_TO_LOG_BUF(...)  do {                              \
+    num_bytes_written = sprintf ( log_buf_ptr, __VA_ARGS__ );                 \
+    if( FD_LIKELY( num_bytes_written>=0 )) log_buf_ptr += num_bytes_written;  \
+ } while(0)
+                                             
+#define FD_LOG_HEXDUMP_DEBUG(a)   do { long _fd_log_msg_now = fd_log_wallclock();           \
+    char const * msg = fd_log_private_hexdump_msg a;                                        \
+    if( msg ) fd_log_private_1( 0, _fd_log_msg_now, __FILE__, __LINE__, __func__, msg );    \
+    } while(0)                                                                              
```

Comment:
So, it looks like the hexdump_msg has the property that calls with edge cases return NULL and a LOG_WARNING is always done in such cases.  This could potentially result in a log message at one level having a different apparent priority than the user expected and this is a potential source of nasty.

On the mild end of the spectrum, some code had a FD_LOG_INFO and was working was fine, LOG implementation was tweaked under the hood to use different limits description limits or what not, FD_LOG_INFO code now actually triggers a FD_LOG_WARNING and suddenly no info present.

Or the uh-oh end of the spectrum, some code does:
```
if( ... detected malicious behavior ... )
  FD_LOG_HEXDUMP_ABORT(( "malicious_packet", pkt_payload, pkt_payload_sz ));
```
Now if there was some innocuous code change that caused HEXDUMP to be unhappy, the ABORT would be demoted to WANRING such that the malicious packet details would not be logged (just a note that HEXDUMP wasn't happy at warning level) and the code would continue processing the malicious packet as though it was not malicious.

Talking this through, pretty clear that the user's log warning level is sacrosanct.

Probably should just have `fd_log_private_hexdump_msg` write the edge case message directly in `log_msg` and return it.  This would have the strict property that it would always return non-NULL and could be compositionally fed directly into `fd_log_private_1` (like the above macros) and the only risk is some loss of detail in the log messages in situations like the uh-oh case above.

Similarly for the macros below.